src/Stratis.Bitcoin.Features.Dns.Tests/GivenADnsFeature.cs

```diff
@@ -281,31 +281,17 @@ public void WhenDnsServerFailsToStart_ThenDnsFeatureRetries()
         public void WhenInitialize_ThenRefreshLoopIsStarted()
         {
             // Arrange.
-            var mockWhitelistManager = new Mock<IWhitelistManager>();
-            mockWhitelistManager.Setup(w => w.RefreshWhitelist()).Verifiable("the RefreshWhitelist method should be called on the WhitelistManager");
-
-            IWhitelistManager whitelistManager = mockWhitelistManager.Object;
-
-            var mockLogger = new Mock<ILogger>();
-            var mockLoggerFactory = new Mock<ILoggerFactory>();
-            mockLoggerFactory.Setup(l => l.CreateLogger(It.IsAny<string>())).Returns(mockLogger.Object);
-            ILoggerFactory loggerFactory = mockLoggerFactory.Object;
-
-            IAsyncLoopFactory asyncLoopFactory = new AsyncLoopFactory(loggerFactory);
-            INodeLifetime nodeLifeTime = new NodeLifetime();
-
-            IDnsServer dnsServer = new Mock<IDnsServer>().Object;
+            this.defaultConstructorParameters.whitelistManager.Setup(w => w.RefreshWhitelist()).Verifiable("the RefreshWhitelist method should be called on the WhitelistManager");
 
             var source = new CancellationTokenSource(3000);
-            var nodeLifetime = new Mock<INodeLifetime>();
-            nodeLifetime.Setup(n => n.StopApplication()).Callback(() => source.Cancel());
-            nodeLifetime.Setup(n => n.ApplicationStopping).Returns(source.Token);
-            INodeLifetime nodeLifetimeObject = nodeLifetime.Object;
+            this.defaultConstructorParameters.nodeLifetime.Setup(n => n.StopApplication()).Callback(() => source.Cancel());
+            this.defaultConstructorParameters.nodeLifetime.Setup(n => n.ApplicationStopping).Returns(source.Token);
+
+            this.defaultConstructorParameters.nodeSettings = new NodeSettings(args: new string[] { $"-datadir={ Directory.GetCurrentDirectory() }" });
```

Comment:
Please pass a network in NodeSettings (a PR is coming that makes this mandatory)

Reply:
ok, i removed all this `new Nodesettings` because was enough to declare it one time in my TextContext constructor